src/NHibernate/Linq/Functions/StringGenerator.cs

```diff
@@ -260,13 +265,9 @@ public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject,
 			else
 				trimWhere = "both";
 
-			string trimChars = "";
-			if (method.GetParameters().Length > 0)
-				foreach (char c in (char[])((ConstantExpression)arguments[0]).Value)
-					trimChars += c;
-
+			var trimChars = ExtractTrimChars(arguments);
 
-			if (trimChars == "")
+			if (trimChars == string.Empty)
```

Comment:
I would rather use `IsNullOrEmpty` here (or since it is never null, eventually `Length == 0`). Comparing string for testing emptiness is not usual.

Reply:
Done